File: app/src/main/java/com/sxshi/retrofit/utils/ParamUtils.java
package com.sxshi.retrofit.utils;

import com.google.gson.Gson;

import java.util.HashMap;
import java.util.Map;

import okhttp3.RequestBody;

/**
 * 公共参数封装
 * 项目公用的请求参数统一在这里处理
 */
public class ParamUtils {

    public static RequestBody getRequestBody(Map<String, String> params) {
        String json = new Gson().toJson(params);

        return RequestBody.create(okhttp3.MediaType.parse("application/json; charset=utf-8"), json);
    }

    public static RequestBody getRequestBody(String attributeValue) {
        HashMap<String, String> params = new HashMap<>();
        params.put("attributeName", attributeValue);//真实项目替换成真实的参数
        String json = new Gson().toJson(params);

        return RequestBody.create(okhttp3.MediaType.parse("application/json; charset=utf-8"), json);
    }
}

File: app/src/main/java/com/sxshi/retrofit/utils/ParamUtils.kt
package com.sxshi.retrofit.utils

import com.google.gson.Gson
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.MultipartBody
import okhttp3.RequestBody
import okhttp3.RequestBody.Companion.asRequestBody
import okhttp3.RequestBody.Companion.toRequestBody
import org.jetbrains.annotations.NotNull
import java.io.File
import java.util.*

/**
 * 公共参数封装
 * 项目公用的请求参数统一在这里处理
 */
object ParamUtils {
    fun getRequestBody(params: Map<String, String>): RequestBody {
        val json = Gson().toJson(params)
        return json.toRequestBody("application/json;charset=utf-8".toMediaType())
    }

    fun getRequestBody(key: String, value: String): RequestBody {
        val params = HashMap<String, String>()
        params[key] = value
        val json = Gson().toJson(params)
        return json.toRequestBody("application/json;charset=utf-8".toMediaType())
    }

    /**
     * MultipartBody.Part
     * 上传文件格式
     */
    fun producePart(@NotNull file: File): MultipartBody.Part {
        val requestFile = file.asRequestBody("multipart/form-data".toMediaType())
        return MultipartBody.Part.createFormData("authFile", file.name, requestFile)
    }

    /**
     * 上传文件 并且携带参数
     */
    fun uploadRequestBody(imageType: String, delta: String, file: File): RequestBody {
        val requestFile = file.asRequestBody("application/octet-stream".toMediaType())
        return MultipartBody.Builder()
                .setType(MultipartBody.FORM)
                .addFormDataPart("imageType", imageType)
                .addFormDataPart("delta", delta)
                .addFormDataPart("authFile", file.getName(), requestFile)
                .build()
    }
}